index.bs

```diff
@@ -10867,41 +10872,50 @@ There must exist an <dfn id="dfn-interface-prototype-object" export>interface pr
-
-As with the [=interface object=],
-the [=interface prototype object=] also has properties that
-correspond to the [=constants=] defined on that interface,
-described in [[#es-constants]].
-
-If the [{{NoInterfaceObject}}] extended attribute was not specified on the interface,
-then the [=interface prototype object=] must also
-have a <code class="idl">constructor</code> data property with attributes
-{ \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
-whose value is a reference to the [=interface object=] for the interface.
 
-<div algorithm="value of the [[Prototype]] internal slot">
+<div algorithm>
 
-    The [=interface prototype object=]
-    for a given interface |A| must have a
-    \[[Prototype]] [=internal slot=] whose value is returned from
-    the following steps:
-
-    1.  If |A| is declared with the [{{Global}}] [=extended attribute=], and
-        |A| [=support named properties|supports named properties=], then
-        return the [=named properties object=]
-        for |A|, as defined in [[#named-properties-object]].
-    1.  Otherwise, if |A| is declared to inherit from another
-        interface, then return the
-        [=interface prototype object=]
-        for the inherited interface.
-    1.  Otherwise, if |A| is declared with the [{{LegacyArrayClass}}]
-        extended attribute, then return {{%ArrayPrototype%}}.
-    1.  Otherwise, return {{%ObjectPrototype%}}.
+    The [=interface prototype object=] for a given [=interface=] |interface| and [=Realm=] |realm|
+    is <dfn lt="create an interface prototype object">created</dfn> as follows:
+
+    1.  Let |proto| be null.
+    1.  If |interface| is declared with the [{{Global}}] [=extended attribute=],
+        and |interface| [=support named properties|supports named properties=],
+        then set |proto| to the [=named properties object=] in |realm| of |interface|.
+    1.  Otherwise, if |interface| is declared to inherit from another interface,
+        then set |proto| to the [=interface prototype object=] in |realm|
+        of that [=inherited interface=].
+    1.  Otherwise, if |interface| is the {{DOMException}} [=interface=],
+        then set |proto| to |realm|.\[[Intrinsics]].[[{{%ErrorPrototype%}}]].
+    1.  Otherwise, if |interface| is declared with the [{{LegacyArrayClass}}] [=extended attribute=],
+        then set |proto| to |realm|.\[[Intrinsics]].[[{{%ArrayPrototype%}}]].
+    1.  Otherwise, set |proto| to |realm|.\[[Intrinsics]].[[{{%ObjectPrototype%}}]].
+    1.  Assert: <a abstract-op>Type</a>(|proto|) is Object.
+    1.  Let |interfaceProtoObj| be [=!=] <a abstract-op>ObjectCreate</a>(|proto|).
+    1.  If |interface| has any [=member=] declared with the [{{Unscopable}}] [=extended attribute=],
+        then:
+        1.  Let |unscopableObject| be the result of performing [=!=]
+            <a abstract-op>ObjectCreate</a>(<emu-val>null</emu-val>).
```

Comment:
The use of null here is a normative change that updates us to match the ES spec. Does it close an open bug? It'll be worth adding tests for :).

Reply:
The null unscopable thing is https://github.com/heycam/webidl/issues/496

Safari already has the right behavior.  

I filed https://bugs.chromium.org/p/chromium/issues/detail?id=811029 (though of course bugs.chromium.org picked up the wrong account _again_, so if that can be changed or the issue deleted I would appreciate that) and https://bugzilla.mozilla.org/show_bug.cgi?id=1437255 which has idlharness tests for this stuff.

I can't file bugs on Edge, unfortunately, so someone else will have to do that.

Reply:
Filed the remaining bugs and consolidated all links to issues in https://github.com/heycam/webidl/issues/496#issuecomment-368589578.